databases/week1/Home work lesson 1.sql

```diff
@@ -0,0 +1,57 @@
+
+-- 1.Find out how many tasks are in the task table
+SELECT count(id) AS TotalTasks
+FROM task;
+
+-- 2.Find out how many tasks in the task table do not have a valid due date
+SELECT count(id) AS TotalTasks
+FROM task
+WHERE task.due_date is NULL;
+
+-- 3.Find all the tasks that are marked as done
+SELECT task.id, task.title, status.name As status
+FROM task 
+JOIN status ON task.status_id = status.id
+WHERE status.name = "Done";
+
+-- 4.Find all the tasks that are not marked as done
+SELECT task.id, task.title, status.name As status
+FROM task 
+JOIN status ON task.status_id = status.id
+WHERE status.name != "Done";
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+order by created desc;
+
+-- 6. Get the single most recently created task
+SELECT *
+ FROM task 
+ ORDER BY created DESC 
+ limit 1;
+ 
+ -- 7. Get the title and due date of all tasks where the title or description contains database
+ SELECT title, due_date
+ FROM task
+ WHERE title LIKE "%database%"
```

Comment:
In mysql the `LIKE` is not case insensitive. Unless you are positive you only want this word with this exact case, you could use `WHERE LOWER(description) LIKE "%database%" OR LOWER(title) LIKE "%database%";`